pkg/ec/lint.go

```diff
@@ -0,0 +1,94 @@
+package ec
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots-lint/pkg/domain"
+	"gopkg.in/yaml.v2"
+)
+
+var ecVersions map[string]bool
+var rwMutex sync.RWMutex
+
+func init() {
+	ecVersions = make(map[string]bool)
+}
+
+func LintEmbeddedClusterVersion(specFiles domain.SpecFiles) ([]domain.LintExpression, error) {
+	lintExpressions := []domain.LintExpression{}
+	// separate multi docs because the manifest can be a part of a multi doc yaml file
+	separatedSpecFiles, err := specFiles.Separate()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to separate multi docs")
+	}
+
+	for _, spec := range separatedSpecFiles {
+		var version string
+		var versionExists bool
+		doc := map[string]interface{}{}
+		if err := yaml.Unmarshal([]byte(spec.Content), &doc); err != nil {
+			return nil, errors.Wrap(err, "failed to unmarshal spec content")
+		}
+		if doc["apiVersion"] == "embeddedcluster.replicated.com/v1beta1" && doc["kind"] == "Config" {
+			if spec, ok := doc["spec"].(map[interface{}]interface{}); ok {
+				version, versionExists = spec["version"].(string)
+			}
+		}
+
+		// if no version exists, continue to next file
+		if !versionExists {
```

Comment:
if the ec config exists without a version, that should be an error as the version is required. we can ignore and continue to next file if the doc isn't an ec config.